client/verta/tests/test_deployment.py

```diff
@@ -579,6 +580,175 @@ def test_dataframe(self, experiment_run, model_for_deployment):
         assert X_train.join(y_train).to_csv(index=False) == six.ensure_str(data_csv)
 
 
+class TestHistogram:
+    @staticmethod
+    def assert_histograms_match_dataframe(histograms, df):
+        """Common assertions for this test suite."""
+        # features match
+        assert set(histograms['features'].keys()) == set(df.columns)
+        # all rows counted
+        assert histograms['total_count'] == len(df.index)
+
+        for feature_name, histogram in histograms['features'].items():
+            series = df[feature_name]
+            histogram_type = histogram['type']
+            histogram_data = histogram['histogram'][histogram_type]
+
+            # all data points counted
+            assert sum(histogram_data['count']) == len(series)
+
+            if histogram_type == "binary":
+                num_false = sum(~series)
+                num_true = sum(series)
+
+                assert histogram_data['count'] == [num_false, num_true]
+            elif histogram_type == "discrete":
+                buckets = histogram_data['bucket_values']
+                counts = histogram_data['count']
+
+                # buckets in ascending order
+                assert buckets == list(sorted(buckets))
+
+                # data within buckets
+                assert all(buckets[0] <= series)
+                assert all(series <= buckets[-1])
+
+                # counts correct
+                for value, count in zip(buckets, counts):
+                    assert sum(series == value) == count
+            elif histogram_type == "float":
+                buckets = histogram_data['bucket_limits']
+                counts = histogram_data['count']
+
+                # buckets in ascending order
+                assert buckets == list(sorted(buckets))
+
+                # data within buckets
+                assert all(buckets[0] <= series)
+                assert all(series <= buckets[-1])
+
+                # counts correct
```

Comment:
Can we add a simple check that the total count from the buckets matches the total too? That's simpler to check than the last assert and a corner case easy to get wrong.

Reply:
Is line 598 what you have in mind?
```python
# all data points counted
assert sum(histogram_data['count']) == len(series)
```

Reply:
Yes!